src/odemis/driver/xt_client.py

```diff
@@ -2279,6 +2288,74 @@ def _moveTo(self, future: futures.Future, pos: Dict[str, float], rel: bool = Fal
                 # know the user that the move is not complete.
                 if future._must_stop.is_set():
                     raise CancelledError()
+
+                # The stage is not moving anymore, however in some rare cases, the server still reports the old
+                # position for a short while (typically < 1s). This can cause all sorts of confusion in the calling
+                # code, as it would seem that the move didn't have any effect, and later on, after the position
+                # is eventually updated that the stage has moved unexpectedly. So, wait until the reported position is
+                # (1) not identical to the starting position and (2) not too far from the target position.
+
+                # The stage is not moving anymore, but we still want to wait until the position has been updated
+                # TODO: base the timeout on the stage movement time estimation (est. *10)
+                timeout = 10  # s
+                expected_end_time = time.time() + timeout  # s
+                check_pos = True
+
+                # TODO: Include rotation and tilt to check stage movement to target position. Test it on a hardware.
+                # The below check to skip the check on stage movement to target position if it includes r ot t
+                # is temporary. It will be removed once the behavior is extended to all the axes and confirmed to work
+                # on a hardware.
+                if 'r' in pos or 't' in pos:
+                    logging.debug("Position requested to move includes t (rx/tilt) or r (rz/rotation). "
+                                  "Reported position accuracy is not checked.")
+                    check_pos = False
+
+                # Get the target position in absolute coordinates
+                if rel:
+                    target_pos = {ax: val + pos[ax] for ax, val in orig_pos.items()}
+                else:
+                    target_pos = pos
+
+                while check_pos:
+                    current_pos = self.parent.get_stage_position()
+                    # Every axis requested to move should have moved (compared to the position before starting)
+                    # if there is some change w.r.t starting position, proceed to check the target position
+
+                    # TODO Check all axes and confirm it works on the hardware
+                    # if not isNearPosition(current_pos=current_pos, target_position=orig_pos, axes=set(orig_pos.keys()),
+                    #                       atol_linear=50e-9, atol_rotation=numpy.radians(0.05)):
```

Comment:
I think the original code, which uses "precisely" not equal is fine. That's because here we just just want to know whether the position has been updated. Even a very tiny change is already a proof that it has been updated.

Reply:
(no need to fix this here, you'll do that once you test that code)

Reply:
I used the original code - pushed just now. To get clarity on the second comment, do we need to fix it during the testing as well or should it be fixed?